File: src/main/java/com/bot/commands/ModerationCommand.java
package com.bot.commands;

import com.bot.utils.CommandCategories;
import net.dv8tion.jda.api.Permission;

public abstract class ModerationCommand extends BaseCommand {
    public ModerationCommand() {
        this.category = CommandCategories.MODERATION;
        this.guildOnly = true;
        this.botPermissions = new Permission[]{Permission.MESSAGE_EMBED_LINKS, Permission.MESSAGE_WRITE, Permission.MESSAGE_ADD_REACTION};
    }
}

File: src/main/java/com/bot/commands/ModerationCommand.kt
package com.bot.commands

import com.bot.utils.CommandCategories
import net.dv8tion.jda.api.Permission

abstract class ModerationCommand : BaseCommand() {
    init {
        this.category = CommandCategories.MODERATION
        this.guildOnly = true
        this.botPermissions = arrayOf(Permission.MESSAGE_EMBED_LINKS, Permission.MESSAGE_WRITE, Permission.MESSAGE_ADD_REACTION)
    }
}
